prov/gni/src/gnix_dom.c

```diff
@@ -65,13 +69,25 @@ static void __domain_destruct(void *obj)
 {
 	int ret = FI_SUCCESS;
 	struct gnix_fid_domain *domain = (struct gnix_fid_domain *) obj;
+	struct gnix_mr_cache_info *info;
+	int i;
 
 	GNIX_TRACE(FI_LOG_DOMAIN, "\n");
 
-	ret = _gnix_close_cache(domain);
-	if (ret != FI_SUCCESS)
-		GNIX_FATAL(FI_LOG_MR, "failed to close memory registration cache\n");
+	for (i = 0; i < 256; i++) {
```

Comment:
256 or a define that says 256, because there is no point to having more than that if we only have 256 distinct ptags.